src/reason-parser/reason_lexer.mll

```diff
@@ -532,7 +532,7 @@ rule token = parse
   | "[|" { LBRACKETBAR }
   | "[<" { LBRACKETLESS }
   | "[>" { LBRACKETGREATER }
-  | "<" uppercase_or_lowercase (identchar | '.') * {
+  | "<" (uppercase identchar* '.')* lowercase identchar* {
```

Comment:
To double check: the main change here is `uppercase_or_lowercase` to `lowercase`, plus some `.` cleanup correct?

The cleanup can be applied to the two cases below too, using I guess
```(uppercase identchar* '.')* uppercase_or_lowercase identchar*```

Reply:
Yes. The change to `lowercase` as the main pattern is to make it only parse lowercase tags (e.g. `<div>`) as `LESSIDENT`. The parser implements the uppercase (opening) tags as a different pattern without `LESSIDENT`. The `.` cleanup was specifically added to allow tags like `<X.x>`, but doesn't support `<X.Y>`. I could add this to the other cases, but                                           they work simply by resetting the lexer to the beginning of the pattern, consuming only up to the `<` token. I assume that the current patterns will run slightly faster, and should work correctly at the parsing level. That said, it may make more sense to switch to `(uppercase identchar* '.')* lowercase identchar*` and `(uppercase identchar* '.')* uppercase identchar*` for the other patterns to make them look more similar and make the patterns more accurate.